File: SrdDatabase/ClientApp/src/models/parish.ts
import { QuotaResults } from './quota';
import { CongregationResults } from './congregation';
import { EventResults } from './event';
import { PaymentResults } from './payment';
import { PagedParameters, PagedResults, Population } from './shared';

export interface Parish {
    id?: number;
    name?: string;
    archdeaconryId?: number;
    archdeaconry?: string;
    numberOfChristians?: number;
    quota?: number;
    balance?: number;
}

export interface ParishDetails {
    parish: Parish;
    population: Population;
    congregationResults: CongregationResults;
    eventResults: EventResults;
    paymentResults: PaymentResults;
    quotaResults: QuotaResults;
}

export interface ParishParameters extends PagedParameters {
    name?: string;
    archdeaconryId?: number;
}

export interface ParishResults extends PagedResults {
    parishes: Parish[];
}

File: SrdDatabase/Models/Parishes/Parish.cs
namespace SrdDatabase.Models.Parishes
{
    public class Parish
    {
        public int Id { get; }

        public string Name { get; }

        public int ArchdeaconryId { get; }

        public string Archdeaconry { get; }

        public int? NumberOfChristians { get; }

        public long Quota { get; }

        public long Balance { get; }
        
        public Parish(
            int id,
            string name,
            int archdeaconryId,
            string archdeaconry,
            int? numberOfChristians,
            long quota,
            long balance)
        {
            Id = id;
            Name = name;
            ArchdeaconryId = archdeaconryId;
            Archdeaconry = archdeaconry;
            NumberOfChristians = numberOfChristians;
            Quota = quota;
            Balance = balance;
        }

        // for Dapper
        public Parish()
        {
        }
    }
}
